encodings/fastlanes/src/for/compress.rs

```diff
@@ -20,10 +20,11 @@ pub fn for_compress(array: PrimitiveArray) -> VortexResult<FoRArray> {
         .compute(Stat::Min)
         .ok_or_else(|| vortex_err!("Min stat not found"))?;
 
-    let nullability = array.dtype().nullability();
+    let dtype = array.dtype().clone();
+    let nullability = dtype.nullability();
     let encoded = match_each_integer_ptype!(array.ptype(), |$T| {
         if shift == <$T>::PTYPE.bit_width() as u8 {
-            assert_eq!(min, Scalar::zero::<$T>(array.dtype().nullability()));
+            assert_eq!(min, Scalar::zero::<$T>(array.dtype().nullability()).into_value());
```

Comment:
Seems a shame to create a scalar with a DType then throw it away? Can you just do `ScalarValue::from(0usize)`?

Reply:
this is me doing silly refactoring